### DISCUSSION TITLE
Why we cant use esp32 wroom bluetooth?

### DISCUSSION THREAD

**Discussion by sajjadarashhh** [nanoframework/Home#1249] [Q&A] [answered]
I jeed to use esp32 wroom bluetooth
But thats not supported by nanoframework
How we can help for this problem??
C++ is used that bluetooth very easy without any complexity

**Comment by alberk8** [comment]
Yes you can.  You need to install the firmware ESP32_BLE_REV0 (rev 1) or ESP32_BLE_REV3 (rev 3).   The rev0 or rev3 refers to the Hardware Revision number.    For example, below I have a Revision 3 (REV3) hardware and this can be seen when flashing the firmware using nanoff.

Example: flashing with Bluetooth firmware

`nanoff --target ESP32_BLE_REV0 --update --serialport COM3`

```
Connected to:
ESP32 (ESP32-D0WD-V3 (revision 3))
Features WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Flash size 8MB unknown from ST (manufacturer 0x32 device 0x16407)
PSRAM: undetermined
Crystal 40MHz
```

**Reply by sajjadarashhh** [reply]
How can i understand to install rev3 or rev1?

**Reply by alberk8** [reply, marked as answer]
Run the command below and let nanoff decide.  The picture below will tell which version of the hardware you are having.

`nanoff  --platform esp32 --update  --serialport COM3 `

![image](https://user-images.githubusercontent.com/12510130/227817298-280a8d25-fa4b-4c6c-8ffc-ae24d56d974d.png)
